test/e2e-go/cli/algoh/expect/algohTimeoutTest.exp

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/expect -f
+set err 0
+log_user 1
+
+
+
+if { [catch {
+    source algohExpectCommon.exp
+
+    set TEST_ALGO_DIR [lindex $argv 0]
+    set TEST_DATA_DIR [lindex $argv 1]
+
+    exec mkdir -p $TEST_ALGO_DIR
+    
+    set TIME_STAMP [clock seconds]
+    set timeout 5
+
+    set TEST_ROOT_DIR $TEST_ALGO_DIR/root
+    set TEST_PRIMARY_NODE_DIR $TEST_ROOT_DIR/Primary
+    set TEST_NODE_DIR $TEST_ROOT_DIR/Node
+    set NETWORK_NAME test_net_expect_$TIME_STAMP
+    set NETWORK_TEMPLATE "$TEST_DATA_DIR/nettemplates/TwoNodes50Each.json"
+    puts "TEST_ALGO_DIR: $TEST_ALGO_DIR"
+
+
+    ::Algoh::EstablishNetwork $NETWORK_NAME $NETWORK_TEMPLATE $TEST_ROOT_DIR
+
+    exec echo > $TEST_NODE_DIR/config.json \{ \"GossipFanout\": 1, \"EndpointAddress\": \"127.0.0.1:0\", \"DNSBootstrapID\": \"\", \"EnableProfiler\": false, \"IncomingConnectionsLimit\": 0, \"RunHosted\": true \}
+    exec echo > $TEST_NODE_DIR/host-config.json \{ \"DeadManTimeSec\": 32, \"StatusDelayMS\": 45000, \"StallDelayMS\": 60000, \"UploadOnError\": true, \"SendBlockStats\": false \}
+    
+    set PRIMARY_ADDR ""
+    spawn cat $TEST_ROOT_DIR/Primary/algod-listen.net
+    expect {
+        -regexp {http://[0-9\.]+:[0-9]+} { regexp -- {[0-9.]+:[0-9]+} $expect_out(0,string) PRIMARY_ADDR; close;}
+        timeout {puts "missed our case"; close; exit 1}
+    }
+
+    puts "regex match: $PRIMARY_ADDR"
+
+    #start hosted node in the background
+    spawn ~/go/bin/algoh -d $TEST_NODE_DIR -p $PRIMARY_ADDR
+    expect {
+        "^Logging to: *" {puts "algoh startup successful"}
+        timeout {puts "algoh failed to start"; close; exit 1}
+    }
+
+    #allow algoh time to put files in Node dir
+    spawn sleep 5
+    set timeout 5
+    expect {
+        timeout {puts "algod should be fully running"; close}
+    }
+
+    #wait until Node approves blocks to the network
+    set timeout 605
+    spawn ~/go/bin/goal node wait -d $TEST_NODE_DIR -w 600
+    expect {
+        eof {puts "successfully communicating with relay node"}
+        "Timed out waiting for node to make progress" {puts "timed out waiting for connection to relay node"; close; exit 1}
+        timeout {puts "should not reached this case"; close; exit 1}
+    }
+
+    spawn ~/go/bin/goal node stop -d $TEST_PRIMARY_NODE_DIR
```

Comment:
shouldn't you use ::Algoh::StopNode ?

Reply:
This procedure doesn't exist but I can add it